### PR TITLE
fix(translation): replace static error message

### PR DESCRIPTION
**Pull request by faust64**
as suggested by @kesselb in nextcloud/server#25490

### CODE REVIEW

File: lib/private/Share20/Manager.php [hunk 1]
```diff
@@ -542,7 +542,8 @@ protected function userCreateChecks(IShare $share) {
 					$this->groupManager->getUserGroupIds($sharedWith)
 			);
 			if (empty($groups)) {
-				throw new \Exception('Sharing is only allowed with group members');
+				$message_t = $this->l->t('Sharing is only allowed with group members');
+				throw new GenericShareException($message_t, $message_t, 404);
```
Comment (kesselb) [inline lib/private/Share20/Manager.php:546]: Please keep \Exception for now. That's easier to backport. We may have different handling for GenericShareException and Exception.

Comment (faust64) [reply lib/private/Share20/Manager.php]: Sure, just throw with that $message_t?

Comment (kesselb) [reply lib/private/Share20/Manager.php]: Yep. Please remove the comment with GenericShare.

Comment (faust64) [reply lib/private/Share20/Manager.php]: Sure.

**Review by kesselb — APPROVED** [review]

**Review by PVince81 — APPROVED** [review]
:+1: makes sense as other user-visible exceptions are translated too

**Comment by MorrisJobke** [comment]
@rullzer Backports failed :/

**Comment by nickvergessen** [comment]
Should we use `GenericShareException`

**Comment by rullzer** [comment]
> Should we use `GenericShareException`

Yeah might make sense now

**Pull request merged**